bin/ch-completion.bash

```diff
@@ -279,8 +281,105 @@ _ch_image_complete () {
 
 # Completion function for ch-run
 #
-_ch_run_completion () {
-    echo "does nothing... yet"
+_ch_run_complete () {
+    local prev
+    local cur
+    local cword
+    local words
+    local strg_dir
+    local extras=
+    #local image
+    _get_comp_words_by_ref -n : cur prev words cword
+
+    strg_dir=$(_ch_find_storage "${words[@]::${#words[@]}-1}")
+    local cli_image
+    local cmd_index=-1
+    _ch_run_image_finder "$strg_dir" "$cword" cli_image cmd_index "${words[@]}"
+
+    # Populate debug log
+    DEBUG "\$ ${words[*]}"
+    DEBUG " storage: dir: $strg_dir"
+    DEBUG " current: $cur"
+    DEBUG " previous: $prev"
+    DEBUG " cli image: $cli_image"
+
+    # Currently, we don’t try to suggest completions if you’re in the “command”
+    # part of the ch-run CLI (i.e. entering commands to be run inside the
+    # container). Implementing this *may* be possible, but that's a complication
+    # I’d prefer to save for a future date.
+    if [[ $cmd_index != -1 && $cmd_index -lt $cword ]]; then
```

Comment:
Honestly, that seems hopelessly complex to me, so I’d frame this comment as a permanent non-feature.